### ISSUE TITLE
"NotFoundException: Entity with the specified id does not exist in the system" - When Running VSTS Load Test on Partitioned Collection

### ISSUE THREAD

**Issue by balajisivaraman** [Azure/azure-cosmos-dotnet-v2#654]
**Describe the bug**

We are getting `Microsoft.Azure.Documents.NotFoundException: Entity with the specified id does not exist in the system` when doing a `Client.CreateDocumentAsync(collectionUri, doc)` operation on our partitioned collection (the details of which are provided below).

This issue does not happen under normal scenarios (such as when our QAs are testing regular scenarios), but only happens when we run a heavy load test on our application. We have also observed that this issue does not happen regularly. The load test succeeds some times and fails with this error at other times.

I'm not sure whether this is related to #457.

**To Reproduce**

We're really not sure since the error does not happen regularly on our setup also.

**Expected behavior**

The `Client.CreateDocumentAsync(collectionUri, doc)` call succeeds without any issues even under heavy load.

**Actual behavior**

The `Client.CreateDocumentAsync(collectionUri, doc)` produces the below stack-trace:

```
Microsoft.Azure.Documents.NotFoundException: Entity with the specified id does not exist in the system.
ActivityId: 84e42a9b-5535-496f-bda7-6d2298546382, 
RequestStartTime: 2019-01-06T18:32:31.5223013Z, Number of regions attempted: 1
, Linux/9 documentdb-netcore-sdk/2.1.3
   at Microsoft.Azure.Documents.AddressResolver.EnsureRoutingMapPresent(DocumentServiceRequest request, CollectionRoutingMap routingMap, DocumentCollection collection)
   at Microsoft.Azure.Documents.AddressResolver.ResolveAddressesAndIdentityAsync(DocumentServiceRequest request, Boolean forceRefreshPartitionAddresses, CancellationToken cancellationToken)
   at Microsoft.Azure.Documents.AddressResolver.ResolveAsync(DocumentServiceRequest request, Boolean forceRefreshPartitionAddresses, CancellationToken cancellationToken)
   at Microsoft.Azure.Documents.AddressSelector.ResolveAddressesAsync(DocumentServiceRequest request, Boolean forceAddressRefresh)
   at Microsoft.Azure.Documents.ConsistencyWriter.WritePrivateAsync(DocumentServiceRequest request, TimeoutHelper timeout, Boolean forceRefresh)
   at Microsoft.Azure.Documents.ConsistencyWriter.WriteAsync(DocumentServiceRequest entity, TimeoutHelper timeout, Boolean forceRefresh)
   at Microsoft.Azure.Documents.ReplicatedResourceClient.InvokeAsync(DocumentServiceRequest request, TimeoutHelper timeout, Boolean isInRetry, Boolean forceRefresh)
   at Microsoft.Azure.Documents.ReplicatedResourceClient.<>c__DisplayClass21_0.<<InvokeAsync>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Documents.BackoffRetryUtility`1.<>c__DisplayClass4_0`1.<<ExecuteAsync>b__1>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Documents.BackoffRetryUtility`1.ExecuteRetryAsync(Func`1 callbackMethod, Func`3 callShouldRetry, Func`1 inBackoffAlternateCallbackMethod, TimeSpan minBackoffForInBackoffCallback, CancellationToken cancellationToken, Action`1 preRetryCallback)
   at Microsoft.Azure.Documents.ShouldRetryResult.ThrowIfDoneTrying(ExceptionDispatchInfo capturedException)
   at Microsoft.Azure.Documents.BackoffRetryUtility`1.ExecuteRetryAsync(Func`1 callbackMethod, Func`3 callShouldRetry, Func`1 inBackoffAlternateCallbackMethod, TimeSpan minBackoffForInBackoffCallback, CancellationToken cancellationToken, Action`1 preRetryCallback)
   at Microsoft.Azure.Documents.BackoffRetryUtility`1.ExecuteAsync[TPolicyArg1](Func`2 callbackMethod, IRetryPolicy`1 retryPolicy, Func`2 inBackoffAlternateCallbackMethod, TimeSpan minBackoffForInBackoffCallback, CancellationToken cancellationToken, Action`1 preRetryCallback)
   at Microsoft.Azure.Documents.ReplicatedResourceClient.InvokeAsync(DocumentServiceRequest request, Func`2 prepareRequestAsyncDelegate, CancellationToken cancellationToken)
   at Microsoft.Azure.Documents.StoreClient.<>c__DisplayClass17_0.<<ProcessMessageAsync>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Documents.StoreClient.ProcessMessageAsync(DocumentServiceRequest request, CancellationToken cancellationToken, IRetryPolicy retryPolicy, Func`2 prepareRequestAsyncDelegate)
   at Microsoft.Azure.Documents.ServerStoreModel.ProcessMessageAsync(DocumentServiceRequest request, CancellationToken cancellationToken)
   at Microsoft.Azure.Documents.Client.DocumentClient.CreateAsync(DocumentServiceRequest request, CancellationToken cancellationToken)
   at Microsoft.Azure.Documents.Client.DocumentClient.CreateDocumentPrivateAsync(String documentCollectionLink, Object document, RequestOptions options, Boolean disableAutomaticIdGeneration, IDocumentClientRetryPolicy retryPolicyInstance, CancellationToken cancellationToken)
   at Microsoft.Azure.Documents.BackoffRetryUtility`1.<>c__DisplayClass1_0.<<ExecuteAsync>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Documents.BackoffRetryUtility`1.ExecuteRetryAsync(Func`1 callbackMethod, Func`3 callShouldRetry, Func`1 inBackoffAlternateCallbackMethod, TimeSpan minBackoffForInBackoffCallback, CancellationToken cancellationToken, Action`1 preRetryCallback)
   at Microsoft.Azure.Documents.ShouldRetryResult.ThrowIfDoneTrying(ExceptionDispatchInfo capturedException)
   at Microsoft.Azure.Documents.BackoffRetryUtility`1.ExecuteRetryAsync(Func`1 callbackMethod, Func`3 callShouldRetry, Func`1 inBackoffAlternateCallbackMethod, TimeSpan minBackoffForInBackoffCallback, CancellationToken cancellationToken, Action`1 preRetryCallback)
   at Microsoft.Azure.Documents.BackoffRetryUtility`1.ExecuteAsync(Func`1 callbackMethod, IRetryPolicy retryPolicy, CancellationToken cancellationToken, Action`1 preRetryCallback)
   at Microsoft.Azure.Documents.Client.DocumentClient.CreateDocumentInlineAsync(String documentsFeedOrDatabaseLink, Object document, RequestOptions options, Boolean disableAutomaticIdGeneration, CancellationToken cancellationToken)
-------
User Code Here, which calls into `Client.CreateDocumentAsync(collectionUri, doc)`
-------
```

**Environment summary**

SDK Version: Microsoft.Azure.DocumentDB.Core/2.1.3
OS Version: Kubernetes Container created from a base image that is Ubuntu 18.04

**Additional context**

- The collection on which this error occurs is used to store Sessions of logged in users and has a generic structure with User Id, a few other user related information, and `SessionId`, which is nothing but a `Guid.New()` generated for each session that has been created.
- `SessionId` is used as our partition key.
- The collection is created with 10000 RUs when we are running load tests.

**Comment by mamoorkhan** [comment]
I am trying to execute the statement

    g.V('abee356e-718f-4f7a-98f1-8107c8ae50e2').drop()

I am getting the following error when I try to execute the statement 

     ServerError: 

    ActivityId : 69ffcc30-4d5c-45eb-abd1-424508f51eeb
    ExceptionType : NotFoundException
    ExceptionMessage :
	    Entity with the specified id does not exist in the system.
	    ActivityId: 69ffcc30-4d5c-45eb-abd1-424508f51eeb, documentdb-dotnet-sdk/2.2.2 Host/64-bit MicrosoftWindowsNT/6.2.9200.0
    Source : mscorlibEntity with the specified id does not exist in the system.
    ActivityId: 69ffcc30-4d5c-45eb-abd1-424508f51eeb, documentdb-dotnet-sdk/2.2.2 Host/64-bit 
    MicrosoftWindowsNT/6.2.9200.0
	BackendStatusCode : NotFound
	BackendActivityId : 69ffcc30-4d5c-45eb-abd1-424508f51eeb
	HResult : 0x80131500

Sometimes it goes away if I have the debugger stopped for a few moments before the following line 

    var resultSet = await SubmitRequest(gremlinClient, gremlinQuery ?? "g.V()");

**Comment by adrianknight89** [comment]
@ealsur This seems similar to #718. I programmatically delete and re-create the collections with the same name and frequently run into this issue. I'm seeing this with `CreateDocumentQuery()` unlike @balajisivaraman's problem with `CreateDocumentAsync`.

@ealsur Was this issue fixed in 1.20.2? I'm using 2.3.0.

**Comment by ealsur** [comment]
@adrianknight89 As far as I know, the issue was addressed for point operations (CreateDocumentAsync for example), not queries.
